packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts

```diff
@@ -93,34 +92,42 @@ export class HostedPluginProcess implements ServerPluginRunner {
         const cp = this.childProcess;
         this.childProcess = undefined;
 
-        const emitter = new Emitter();
+        const waitForTerminated = new Deferred<void>();
         cp.on('message', message => {
-            emitter.fire(JSON.parse(message));
-        });
-        const rpc = new RPCProtocolImpl({
-            onMessage: emitter.event,
-            send: (m: {}) => {
-                if (cp.send) {
-                    cp.send(JSON.stringify(m));
-                }
+            const msg = JSON.parse(message);
+            if ('type' in msg && msg.type === MessageType.Terminated) {
+                waitForTerminated.resolve();
             }
         });
-        const hostedPluginManager = rpc.getProxy(MAIN_RPC_CONTEXT.HOSTED_PLUGIN_MANAGER_EXT);
-        hostedPluginManager.$stop().then(() => {
-            emitter.dispose();
-            this.killProcessTree(cp.pid);
-        });
+        cp.send(JSON.stringify({ type: MessageType.Terminate }));
+        // give 10 seconds to terminate gracefully and then kill the process
```

Comment:
probably allow to be configurable

Reply:
Should it be possible to disable timeout with 0?

Reply:
yes, might be interesting 👍